raiden/blockchain/abi.py

```diff
@@ -115,63 +91,74 @@ def contract_checksum(contract_path):
-CHANNELCLOSED_EVENTID = event_id(*get_eventname_types(CHANNELCLOSED_EVENT))
-
-CHANNELSECRETREVEALED_EVENT = get_event(NETTING_CHANNEL_ABI, 'ChannelSecretRevealed')
-CHANNELSECRETREVEALED_EVENTID = event_id(*get_eventname_types(CHANNELSECRETREVEALED_EVENT))
-
-CHANNELSETTLED_EVENT = get_event(NETTING_CHANNEL_ABI, 'ChannelSettled')
-CHANNELSETTLED_EVENTID = event_id(*get_eventname_types(CHANNELSETTLED_EVENT))
-
-REGISTRY_TRANSLATOR = ContractTranslator(REGISTRY_ABI)
-CHANNEL_MANAGER_TRANSLATOR = ContractTranslator(CHANNEL_MANAGER_ABI)
-NETTING_CHANNEL_TRANSLATOR = ContractTranslator(NETTING_CHANNEL_ABI)
+class ContractManager():
+
+    def __init__(self):
+        self.is_instantiated = False
+
+    def instantiate(self):
+        self.human_standard_token_compiled = get_static_or_compile(
+            get_contract_path('HumanStandardToken.sol'),
+            'HumanStandardToken',
+            combined='abi',
+        )
+
+        self.channel_manager_compiled = get_static_or_compile(
+            get_contract_path('ChannelManagerContract.sol'),
+            'ChannelManagerContract',
+            combined='abi',
+        )
+
+        self.endpoint_registry_compiled = get_static_or_compile(
+            get_contract_path('EndpointRegistry.sol'),
+            'EndpointRegistry',
+            combined='abi',
+        )
+
+        self.netting_channel_compiled = get_static_or_compile(
+            get_contract_path('NettingChannelContract.sol'),
+            'NettingChannelContract',
+            combined='abi',
+        )
+
+        self.registry_compiled = get_static_or_compile(
+            get_contract_path('Registry.sol'),
+            'Registry',
+            combined='abi',
+        )
+
+        self.is_instantiated = True
+
+    def get_abi(self, contract_name):
+        if not self.is_instantiated:
+            self.instantiate()
+
+        compiled = getattr(self, '{}_compiled'.format(contract_name))
+        return compiled['abi']
+
+    def get_event_id(self, event_name):
+        """ Not really generic, as it maps event names to events of specific contracts,
+        but it is good enough for what we want to accomplish.
+        """
+        if event_name == 'TokenAdded':
+            event = get_event(self.get_abi('registry'), event_name)
+        elif event_name == 'ChannelNew':
+            event = get_event(self.get_abi('channel_manager'), event_name)
+        elif event_name == 'ChannelNewBalance':
+            event = get_event(self.get_abi('netting_channel'), event_name)
+        elif event_name == 'ChannelClosed':
+            event = get_event(self.get_abi('netting_channel'), event_name)
+        elif event_name == 'ChannelSecretRevealed':
+            event = get_event(self.get_abi('netting_channel'), event_name)
+        elif event_name == 'ChannelSettled':
+            event = get_event(self.get_abi('netting_channel'), event_name)
+        else:
+            raise ValueError('Unknown event: {}'.format(event_name))
+
```

Comment:
Hm, interesting... Do you remember the argument what would make dict usage here unpythonic?
I feel a dict would make the intent much clearer and also avoid the repetitive boilerplate.